### ISSUE TITLE
[IMPR] Introduce JSII to support more platforms

### ISSUE THREAD

**Issue by szikszail** [gherking/gherking#60]
## Summary
Assess the possibility of introducing [JSII](https://github.com/aws/jsii), to release GherKing packages in JS, PY, Java, Go, etc.

**Comment by kornel-filep** [comment]
I investigated more about the usage of JSII in gherking, and here is what I found:

- JSII is able to create polyglot libraries, it supports a lot of different languages.
- JSII puts a lot of restrictions on what you can use in typescript.
- JSII puts a lot of restrictions on interfaces and inheritance.

I tried out the tool with the gherkin-ast library, and the first obstacle was that I needed to override the tsconfig.json file with the one that JSII generates. This resulted in type errors, I needed to change a lot of the typing in the library. 
After this one was handled, the other roadblock is that we need to address a lot of the restrictions that come with JSII: https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/
At this point i saw 50+ errors in console which are related to these.

So in conclusion:
Introducing JSII to Gherking is for sure possible, but there is a significant time cost. To introduce it, you need to refactor a lot inside the library.

**Comment by szikszail** [comment]
Adding JSII would mean a vast refactor in our API, moving this to V3, where we can consider this.
